### DISCUSSION TITLE
How to delay Dag Run?

### DISCUSSION THREAD

**Discussion by Fthayy** [apache/airflow#33068] [Q&A]
Airflow 2.5.1 
[
![Ekran Alıntısıx](https://github.com/apache/airflow/assets/111737897/977c4a2a-c657-4a1c-b4bd-943ab7aa8ca0)
](url)

I want to delay the Dag Runs. I'm trying to do it with TimeTable.I delayed the Data Interval Start and Data  Interval End but dag worked before them. What I am doing wrong? I found the #28157 but I couldn't understand exactly.

**Comment by potiuk** [comment]
I couldn't understand what you want to do - it's confuxing what you are asking for

**Reply by Fthayy** [reply]
For example if time is earlier than 5pm It'll work directly but if time is later than 5pm it needs to work at midnight. So I delayed Data Interval with timetable but it didn't work. I want to know how can I delay the dag_run.

**Reply by potiuk** [reply]
I have no idea what you are asking for, and what you did to mitigate it. if you want to have conditional behaviiour in your dags you can use various branch operators and sensors (that for example wait for a given time). Look at the docs for those before coming here to see if they solve your problem, but I still have no idea what you are asking for.

**Reply by potiuk** [reply]
Example explaining it in details and rarther than telling "I used timetable to solve it"  showing what you tried to do might help in explaining your point more clearly.

**Reply by potiuk** [reply]
BTW. The more time you spend on clearly describing your problem the more likely it is that someone will want to help you.

**Reply by Fthayy** [reply]
Appreciate for your feedback. 

```
from datetime import timedelta
from pendulum import UTC, DateTime
from airflow.plugins_manager import AirflowPlugin
from airflow.timetables.base import DagRunInfo, DataInterval, TimeRestriction, Timetable
from datetime import datetime

class AfterWorkdayTimetable(Timetable):
    def infer_manual_data_interval(self, run_after: DateTime) -> DataInterval:
        start = datetime.now().replace(tzinfo=UTC)
        return DataInterval(start=start+timedelta(days=6), end=(start + timedelta(days=7)))
    def next_dagrun_info(self,*,last_automated_data_interval: DataInterval | None,restriction: TimeRestriction,) -> DagRunInfo | None:
        return None  
class WorkdayTimetablePlugin(AirflowPlugin):
    name = "workday_timetable_plugin"
    timetables = [AfterWorkdayTimetable]
```
my goal here was to assign manually triggered dags directly to 6 days ahead. Data Interval Start and End updated as you can see in screenshot but dag didn't delayed.

**Reply by potiuk** [reply]
I still don't know wha tyou want to achieve. If you want to trigger dags manually and let them start 6 days later this is not possible. You can trigger a dag and let it wait with the sensor for 6 days using Timedelta sensor as part of the dag, but you cannot trigger dag and way for 6 days. You can write an external service tha twill run trigger API call in 6 days if you want.

**Reply by Fthayy** [reply]
Thanks again for the answer. I thought it might be possible, I'll try to do it a different way
